src/lib/operators/table_scan/is_null_table_scan_impl.cpp

```diff
@@ -13,84 +13,77 @@
 
 namespace opossum {
 
-IsNullTableScanImpl::IsNullTableScanImpl(const std::shared_ptr<const Table>& in_table, const ColumnID base_column_id,
+IsNullTableScanImpl::IsNullTableScanImpl(const std::shared_ptr<const Table>& in_table, const ColumnID column_id,
                                          const PredicateCondition& predicate_condition)
-    : BaseSingleColumnTableScanImpl{in_table, base_column_id, predicate_condition} {
+    : AbstractSingleColumnTableScanImpl{in_table, column_id, predicate_condition} {
   DebugAssert(predicate_condition == PredicateCondition::IsNull || predicate_condition == PredicateCondition::IsNotNull,
               "Invalid PredicateCondition");
 }
 
 std::string IsNullTableScanImpl::description() const { return "IsNullScan"; }
 
-void IsNullTableScanImpl::handle_segment(const ReferenceSegment& base_segment,
-                                         std::shared_ptr<SegmentVisitorContext> base_context) {
-  auto context = std::static_pointer_cast<Context>(base_context);
-  BaseSingleColumnTableScanImpl::handle_segment(base_segment, base_context);
+std::shared_ptr<PosList> IsNullTableScanImpl::scan_chunk(const ChunkID chunk_id) const {
+  const auto& chunk = _in_table->get_chunk(chunk_id);
+  const auto& segment = chunk->get_segment(_column_id);
 
-  const auto& pos_list = *base_segment.pos_list();
+  auto matches = std::make_shared<PosList>();
 
-  // Additionally to the null values in the referencED segment, we need to find null values in the referencING segment
-  if (_predicate_condition == PredicateCondition::IsNull) {
-    for (ChunkOffset chunk_offset{0}; chunk_offset < pos_list.size(); ++chunk_offset) {
-      if (pos_list[chunk_offset].is_null()) context->_matches_out.emplace_back(context->_chunk_id, chunk_offset);
-    }
-  }
+  _on_scan(*segment, chunk_id, *matches, nullptr);
+
+  return matches;
+}
+
+void IsNullTableScanImpl::_on_scan(const BaseSegment& segment, const ChunkID chunk_id, PosList& matches,
+                                   const std::shared_ptr<const PosList>& position_filter) const {
```

Comment:
Can we pass the position_filter as `const PosList&`? Just trying to safe some chars...

Reply:
Nope, because the iterables want a pointer (which they can check for `nullptr`)